src/main/java/be/syntra/devshop/DevshopFront/controllers/SearchController.java

```diff
@@ -47,75 +48,73 @@ public SearchController(
     @GetMapping("/")
     public String showSearchBarResult(@RequestParam String searchRequest, Model model) {
         searchService.setSearchRequest(searchRequest);
-        List<Product> productList = applySearch(true);
+        ProductListAndMinMaxPrice productListAndMinMaxPrice = applySearch(true);
+        List<Product> productList = productListAndMinMaxPrice.getProducts();
+        searchService.setPriceFilters(productListAndMinMaxPrice.getSearchResultMinPrice(),productListAndMinMaxPrice.getSearchResultMaxPrice());
         return setTemplateModel(model, productList);
     }
 
     @GetMapping("/pricelow/")
     public String searchPriceLow(@RequestParam String priceLow, Model model) {
         searchService.setPriceLow(new BigDecimal(priceLow));
-        List<Product> productList = applySearch(true);
+        ProductListAndMinMaxPrice productListAndMinMaxPrice = applySearch(true);
+        List<Product> productList = productListAndMinMaxPrice.getProducts();
         return setTemplateModel(model, productList);
     }
 
     @GetMapping("/pricehigh/")
     public String searchPriceHigh(@RequestParam String priceHigh, Model model) {
         searchService.setPriceHigh(new BigDecimal(priceHigh));
-        List<Product> productList = applySearch(true);
+        ProductListAndMinMaxPrice productListAndMinMaxPrice = applySearch(true);
```

Comment:
you don't need the ProductList here, so it could just be: 
List<Product> productList = applySearch(true).getProducts();

Reply:
corrected by using ProductList and moved to applySearch method